### DISCUSSION TITLE
Context Menu

### DISCUSSION THREAD

**Discussion by samcarey** [emilk/egui#233] [Q&A] [answered]
I'm trying to create a context menu. Is there an example for this? Here's what i tried:

```
let response = ui.label("test");
if response.secondary_clicked() {
    *last_click = ui.input().pointer.interact_pos();
}
popup::show_tooltip_at(ctx, ui.id(), *last_click, |ui| {
    if ui.button("Button").clicked() {
        println!("Clicked");
    };
});
```
Where `last_click` is an `Option<Pos2>` stored in my struct implementing `epi::App`.
The button doesn't seem to react when I hover or click.

**Comment by emilk** [comment, marked as answer]
The problem here is that labels don't respond to clicks by default. You can use this: https://docs.rs/egui/0.10.0/egui/struct.Response.html#method.interact

**Reply by emilk** [reply]
`show_tooltip_at` won't work for you either - tooltips just close when you click them. You can use `popup_below_widget` instead. This is a somewhat under-developed part of `egui`.

**Reply by samcarey** [reply]
> 
> 
> `show_tooltip_at` won't work for you either - tooltips just close when you click them. You can use `popup_below_widget` instead. This is a somewhat under-developed part of `egui`.

Ok I was able to make my own function based on `popup_below_widget` that does what I want. Thanks!
